Fix libfido2.framework structure for Xcode 26 compatibility

aerickson14 opened the pull request:
- Convert libfido2.framework from shallow bundle to deep bundle structure
- Move Info.plist to Versions/Current/Resources/Info.plist as required by Xcode 26
- Create proper Versions/A directory structure with Current symlink
- Add symlinks in framework root pointing to versioned resources

This fixes the archiving error:
'contains Info.plist, expected Versions/Current/Resources/Info.plist since the platform does not use shallow bundles'

Resolves compatibility issues when archiving apps with Xcode 26.

aerickson14 committed:
Fix libfido2.framework structure for Xcode 26 compatibility

- Convert libfido2.framework from shallow bundle to deep bundle structure
- Move Info.plist to Versions/Current/Resources/Info.plist as required by Xcode 26
- Create proper Versions/A directory structure with Current symlink
- Add symlinks in framework root pointing to versioned resources

This fixes the archiving error:
'contains Info.plist, expected Versions/Current/Resources/Info.plist
since the platform does not use shallow bundles'

Resolves compatibility issues when archiving apps with Xcode 26.

MattKiazyk commented:
@aerickson14 that fixes the one issue - but still getting code signing errors. 

If I add the 3 framework in `Link Binary with libraries` build phase it gets passed it.

I'm guessing however that I would still have the same problem using the libfido2swift package outside with the same issue. Perhaps a workaround for myself is to copy/paste the library right into Xcodes and that might work.

aerickson14 committed:
Move LICENSE file to Resources directory to fix code signing

- LICENSE files should be in Resources, not in the main framework directory
- Fixes code signing error where LICENSE was treated as a binary to sign

MattKiazyk commented:
@aerickson14 this works I think - as I grabbed your branch for xcodes and it's building. 

@kinoroy when you have a minute can you merge this in.

kinoroy commented:
Thank you so much @aerickson14 for helping out with this 🙌

@MattKiazyk I'll merge this and make a new release soon
